src/llmcompressor/pipelines/sequential/pipeline.py

```diff
@@ -0,0 +1,77 @@
+from contextlib import nullcontext
+from typing import List
+
+import torch
+import torch.utils.data.dataloader
+import tqdm
+from compressed_tensors.utils import get_execution_device
+
+from llmcompressor.modifiers.utils.hooks import HooksMixin
+from llmcompressor.pipelines.cache import IntermediatesCache
+from llmcompressor.pipelines.sequential.helpers import trace_subgraphs
+from llmcompressor.utils.helpers import calibration_forward_context
+
+__all__ = ["run_pipeline"]
+
+
+def run_pipeline(
+    model: torch.nn.Module,
+    sequential_targets: List[str],
+    ignore: List[str],
+    dataloader: torch.utils.data.DataLoader,
+    propagate_error: bool,
+):
+    """
+    Run a sequential data pipeline.
+    1. The model is partitioned into subgraphs according to `sequential_targets`
+    2. Data passes through each subgraph sequentially. If `propagate_error` is enabled,
+        then data is passed through each subgraph twice, once to trigger calibration
+        hooks, then a second time in order to capture activations after quantization
+        has occurred through the hooks.
+    3. The intermediate activations between each subgraph are cached and offloaded to
+        the cpu between each batch in order to save memory
+
+    This pipeline requires that the model be tracable with respect to data from the
+    data loader. This may be an issue for vision language models with vision datasets,
+    due to specialized input processing in the model.
+
+    In the event that tracing fails, a torch.fx.proxy.TraceError will be raised. A model
+    can be made tracable by wrapping the untracable functions (see
+    llmcompressor.transformers.tracing)
+    """
+    # trace subgraphs
+    sample_input = next(iter(dataloader))
+    subgraphs = trace_subgraphs(model, sample_input, sequential_targets, ignore)
+
+    with calibration_forward_context(model):
+        # prepare intermediates cache
+        model_device = get_execution_device(model)
+        intermediates = IntermediatesCache.from_dataloader(dataloader, model_device)
+
+        num_subgraphs = len(subgraphs)
+        for subgraph_index, subgraph in enumerate(subgraphs):
+            # prepare tqdm description texts
+            calib_desc = f"({subgraph_index + 1}/{num_subgraphs}): Calibrating"
+            prop_desc = f"({subgraph_index + 1}/{num_subgraphs}): Propagating"
+
+            # compile subgraph forward function
+            forward_function = subgraph.compile_forward()
+
+            if propagate_error:
```

Comment:
same as other pipeline, let's remove case for `propagate_error=False`

Reply:
Good idea